osx/src/platform_osx.mm

```diff
@@ -58,57 +58,55 @@ bool isContinuousRendering() {
 
 }
 
-NSString* resolveResourcePath(const char* _path) {
+NSString* resolvePath(const char* _path, PathType _type) {
 
     if (s_resourceRoot == NULL) {
         setResourceRoot(".");
     }
 
-    return [s_resourceRoot stringByAppendingPathComponent:[NSString stringWithUTF8String:_path]];
+    NSString* path = [NSString stringWithUTF8String:_path];
 
+    switch (_type) {
+    case PathType::internal:
+        return [[[NSBundle mainBundle] resourcePath] stringByAppendingPathComponent:path];
+    case PathType::resource:
+        return [s_resourceRoot stringByAppendingPathComponent:path];
+    case PathType::absolute:
+        return path;
+    }
 }
 
-std::string stringFromResource(const char* _path) {
+std::string stringFromFile(const char* _path, PathType _type) {
     
-    NSString* path = resolveResourcePath(_path);
+    NSString* path = resolvePath(_path, _type);
     NSString* str = [NSString stringWithContentsOfFile:path
                                               encoding:NSASCIIStringEncoding
                                                  error:NULL];
-    
+
     if (str == nil) {
-        LOGW("Failed to read file at path: %s", _path);
+        LOGW("Failed to read file at path: %s", [path UTF8String]);
         return std::move(std::string());
```

Comment:
This doesn't come from this PR, but it might be the right time to remove the `std::move` and let the compiler do return value optimization if needed. This would also match the new `osx` way of returning strings.